File: api/add_articles_to_pinecone.py
import csv
import os

from langchain_openai import OpenAIEmbeddings
from pinecone import Pinecone, ServerlessSpec
from langchain_pinecone import PineconeVectorStore
import dotenv

dotenv.load_dotenv()



model_name = "text-embedding-3-small"  
embeddings = OpenAIEmbeddings(  
    model=model_name,  
    openai_api_key=os.environ.get("OPENAI_API_KEY")  
)  
print(os.environ.get("PINECONE_API_KEY"))

pc = Pinecone(api_key=os.environ.get("PINECONE_API_KEY"))

index_name = "sanfranfoodarticle"

if index_name not in pc.list_indexes().names():
    pc.create_index(
        name=index_name,
        dimension=1536, 
        metric="cosine", 
        spec=ServerlessSpec(
            cloud="aws", 
            region="us-east-1"
        ) 
    ) 




namespace = "wondervector5000"


    
idCount = 0
with open("my_scraped_articles.csv", mode='r', newline='', encoding='utf-8') as file:
        reader = csv.DictReader(file)
        for row in reader:
            print(row['Title'])

            docsearch = PineconeVectorStore.from_texts(
                texts=[row['Text']],
                metadatas=[{"title":row['Title'], "text":row["Text"], "source":row["Source"]}],
                index_name=index_name,
                embedding=embeddings, 
                namespace=namespace 
            )
            # collection.add(documents=[row['Text']], ids=[str(idCount)],metadatas={"title":row['Title'], "text":row["Text"]})            
            print("finished with article")
            idCount+=1

print("Finished")

File: web/add_articles_to_pinecone.js
import { Pinecone } from "@pinecone-database/pinecone";
import { EmbeddingModel, FlagEmbedding } from "fastembed";
import { readFileSync } from "fs";
import * as csv from "@fast-csv/parse";

const pc = new Pinecone({
  apiKey: "",
});

const embeddingModel = await FlagEmbedding.init({
  model: EmbeddingModel.AllMiniLML6V2,
});

const fileContent = readFileSync("./my_scraped_articles.csv");

await pc.createIndex({
  name: "quickstart",
  dimension: 8, // Replace with your model dimensions
  metric: "euclidean", // Replace with your model metric
  spec: {
    serverless: {
      cloud: "aws",
      region: "us-east-1",
    },
  },
});
